### ISSUE TITLE
Support JsonStringEnumConverter

### ISSUE THREAD

**Issue by Vulthil** [nenoNaninu/Tapper#64]
It would be nice if the transpilation from C# Enums to TypeScript Enums supported the `[JsonConverter(typeof(JsonStringEnumConverter))]` Attribute from the `System.Text.Json.Serialization` namespace.

This could be specified as an option flag to be opt-in.

An  example could be
```csharp
[JsonConverter(typeof(JsonStringEnumConverter))]
[TranspilationSource]
public enum SomeEnum
{
    SomeValue,
    SomeOtherValue
}
```

which at the moment is transpiled to 
```ts
/** Transpiled from SomeNamespace */
export enum SomeEnum {
    SomeValue = 0,
    SomeOtherValue = 1,
}
```
The corresponding JsonSerialized value being used in a SignalR hub will use the string values of the enum value, hence a mismatch between the types.
Meaning a C# object like
```csharp
public class SomeClass {
  public SomeEnum MyEnum {get;set;} = SomeEnum.SomeValue
}
```
will be serialized to by the SignalR hub
```json
{
 "MyEnum": "SomeValue"
}
``` 

This makes the enum behave wrong when being transpiled.

A solution could be to have `Tapper` generate something like the following
```ts
/** Transpiled from SomeNamespace */
export enum SomeEnum {
    SomeValue = "someValue", // or "SomeValue"
    SomeOtherValue = "someOtherValue", // or "SomeOtherValue"
}
```

**Comment by nenoNaninu** [comment]
It is a good idea to support some attributes in `System.Text.Json`.

- `[JsonPropertyName]`
- `[JsonIgnore]`
- `[JsonConverter(typeof(JsonStringEnumConverter))]`

**Comment by nenoNaninu** [comment]
I noticed that it is a bad idea to change the behavior depending on whether the JsonConverter attribute is annotated. This is because there are cases where the JsonSerializerOptions are registered instead of the type definition.

---
Look up attributes in `System.Text.Json` using flags such as `--use-json-attribute`.

target: `[JsonPropertyName]` / `[JsonIgnore]`

```
tapper --project xxx --output yyy --use-json-attribute
```
---

Change the value of an enum with a option such as `--enum-value`.

target: `JsonStringEnumConverter`.

```
tapper --project xxx --output yyy --enum-value default (underlying-type, integer)
tapper --project xxx --output yyy --enum-value name
tapper --project xxx --output yyy --enum-value nameAsCamel
```

These are still in the idea stage (not yet finalized).

**Comment by thisisclint21** [comment]
Also what about the possibility to export enum as type? 

like:
```ts
type Statuses = "Paid" | "NotPaid";
```

**Comment by thisisclint21** [comment]
Would be really great @Vulthil @nenoNaninu to have enum as names!

**Comment by nenoNaninu** [comment]
I released v1.7, which added the feature to transpile `enum` to `string enum` and `string union`.
For more details, [please see docs](https://github.com/nenoNaninu/Tapper#enum-style).
